### ISSUE TITLE
eth_getFilterChanges does not seem to return anything

### ISSUE THREAD

**Issue by tzapu** [paritytech/parity#6911]
> I'm running:
>
> - **Parity version**: 1.7.2 1.7.7 1.8.2
> - **Operating system**: Linux

over a web-sockets connection I tried to get pending transactions 
```
> {"method":"eth_newPendingTransactionFilter","params":[],"id":1,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":"0x0","id":1}
> {"method":"eth_getFilterChanges","params":["0x0"],"id":1,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":[],"id":1}
> {"method":"eth_getFilterChanges","params":["0x0"],"id":1,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":[],"id":1}
> {"method":"eth_getFilterChanges","params":["0x0"],"id":1,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":[],"id":1}
> {"method":"eth_getFilterChanges","params":["0x0"],"id":1,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":[],"id":1}
> {"method":"eth_getFilterChanges","params":["0x0"],"id":1,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":[],"id":1}
> {"method":"eth_getFilterChanges","params":["0x0"],"id":1,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":[],"id":1}
> {"method":"eth_getFilterChanges","params":["0x0"],"id":1,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":[],"id":1}
> {"method":"eth_getFilterChanges","params":["0x0"],"id":1,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":[],"id":1}
> {"method":"eth_getFilterChanges","params":["0x0"],"id":1,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":[],"id":1}
> {"method":"eth_getFilterChanges","params":["0x0"],"id":1,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":[],"id":1}
> {"method":"eth_getFilterChanges","params":["0x0"],"id":1,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":[],"id":1}
> {"method":"eth_getFilterChanges","params":["0x0"],"id":1,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":[],"id":1}
> {"method":"eth_getFilterChanges","params":["0x0"],"id":100,"jsonrpc":"2.0"}
< {"jsonrpc":"2.0","result":[],"id":100}
```
---

nothing is ever returned. Did I use it wrong or does it simply not work?
`parity_pendingTransactions` returns plenty of transactions

cheers

**Comment by Office-Julia** [comment]
Hello. Could you please share the following information:
1. Have you tried turning it off and on again?
2. Are you fully synchronized?
3. How did you install Parity? (binary, installer, from source, which branch?)
4. What flags are you running Parity with?
5. Do you use any config file?
6. Please, share the exact error message!
7. Is there anything in the logs?
8. Are you connected to any peers?
9. What's your time? Is it synchronized? time.is
10. Did you answer all my questions yet?

**Comment by tzapu** [comment]
some more info, this seems to be more timing related
if the request to get filter changes is made fast enough after setting the filter it doesn't seem to return anything, but at some point it starts returning the pendings.

maybe this is just a question of updating the docs than a real issue
I read this like this, so please correct me if i am wrong:
calling `eth_newPendingTransactionFilter` means all pending transactions starting when this was called so in effect, **calling `eth_getFilterChanges` returns all pending transactions since last poll or `eth_newPendingTransactionFilter`**

**Comment by debris** [comment]
fixed in #8341
